Szilidan relativepath

szilidaniel opened the pull request:
@hegyhati Az én részem késznek tekinthető viszont van annyi változás az eredetihez képest, hogy a többieknek hozzá kelljen kicsit igazítani, ezért jó lenne ezt felrakni először.

szilidaniel committed:
cd update (relative&absolute path)

cd mostmár jól működik (eddig egy break kimaradt) Mielőtt nekifutnék a többi függvénynek ezt pusholom, hogy legyen egy pont amire visszaállhatok, ha gáz van.

szilidaniel committed:
Feature 1 complete

A feladatot teljesíti bár van még mit javítani rajta

szilidaniel committed:
Remove extra lines

Ez véletlen benne maradt.

oliverosz commented on main.cpp:
```diff
@@ -24,44 +37,62 @@ int main() {
 				cin >> dirname;
-				d.rmForce(dirname, CurrentFolder);
+				vector<string> path;
+				if (dirname.find_first_of("/") < 4294967290) {
```
`string::npos`

oliverosz commented on main.cpp:
```diff
@@ -11,6 +11,19 @@ bool validcommand(string command) {
+vector<string> split(string path) {
+	vector<string> directories;
+	int cut = 0;
+	while (cut < 4294967290) {
```
`string::npos` constant should be used

oliverosz commented on main.cpp:
```diff
@@ -24,44 +37,62 @@ int main() {
+				if (dirname.find_first_of("/") < 4294967290) {
+					path = split(dirname);
+					path.pop_back();
+					dirname = dirname.substr(dirname.find_last_of("/") + 1, dirname.size());
```
this is the same string that was deleted in the previous line from the end of `path`, isn't it?

szilidaniel replied:
Így van. Igazából lehetett volna a vektor utolsó eleme is, de így is ugynaz az eredmény.

oliverosz reviewed (approved):
Ok, lenne min szépíteni, de rendben van. Nézz utána az "arrow anti-pattern" fogalmának, és próbálj leszokni róla!

szilidaniel committed:
CircleCiUpdate

CircleCI tesztpróba

szilidaniel committed:
CircleCI Proba2

szilidaniel committed:
CircleCI Proba3

szilidaniel committed:
CircleCI Proba4

szilidaniel committed:
CircleCI Proba5

szilidaniel committed:
CircleCI Proba6

Még látok reményt, hogy egyszer jó lesz.

szilidaniel committed:
CircleCI Proba7

szilidaniel committed:
CircleCI Proba8

Pedig tényleg átnéztem jól írom-e be.

szilidaniel committed:
Bugfix

A nagy számokat lecseréltem egy elegánsabb megoldásra, valamint mefoldottam, hogy ne kelljen kilépés után még egy input adatot megadni.

szilidaniel commented:
@oliverosz @hegyhati  a kért módosításokat megtettem + CircleCI teszttel kiegészítettem

oliverosz commented:
@szilidaniel javítsd ki a conflictot, addig nem lehet mergelni

szilidaniel committed:
add echo from vasaszs

szilidaniel committed:
Merge conflict solved

Visszaírtam mindent amivel baj volt.

szilidaniel commented:
@oliverosz Jobbat nem tudtam így a Zsuzsa részét beleraktam. Most elvileg jó kell, hogy legyen.

oliverosz reviewed (approved):
Ok. Amúgy a mastert kellett volna bemergelni ebbe a branchbe, hogy ide is bekerüljenek az ott lévő változtatások. És merge közben fel lehet oldani a conflictokat.

Majd az echonak is csináljatok teszteket, de az mehet másik PR-be.

@hegyhati mergelheted.
